Packet++/header/WireGuardLayer.h

```diff
@@ -0,0 +1,253 @@
+#pragma once
+
+#include "Layer.h"
+#include "IpAddress.h"
+#include "MacAddress.h"
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+
+	/**
+	 * WireGuard message types
+	 */
+	enum class WireGuardMessageType
+	{
+		HandshakeInitiation = 1,  ///< Handshake Initiation message
+		HandshakeResponse = 2,    ///< Handshake Response message
+		CookieReply = 3,          ///< Cookie Reply message
+		TransportData = 4         ///< Transport Data message
+	};
+
+	/**
+	 * @struct wg_common_header
+	 * Represents the common header for all WireGuard message types
+	 */
+#pragma pack(push, 1)
+	struct wg_common_header
+	{
+		/** Message type field */
+		uint8_t messageType;
+		/** Reserved field (3 bytes) */
+		uint8_t reserved[3];
+	};
+#pragma pack(pop)
+
+	/**
+	 * @struct wg_handshake_initiation
+	 * Represents the Handshake Initiation message
+	 */
+#pragma pack(push, 1)
+	struct wg_handshake_initiation
+	{
+		wg_common_header common;               ///< Common header for all WireGuard messages
```

Comment:
`wg_handshake_initiation` can also inherit from `wg_common_header`. Ditto in the next structs